### PR TITLE
Add New Pet Mod of the Season (VaporeMons) & Leader's Choice (Modern Gen 1)

### PR DESCRIPTION
**Pull request by Yoshiblaze**

### CODE REVIEW

**Comment by Yoshiblaze** [comment]
The next PMOTS and Leader's Choice were decided through a vote here https://www.smogon.com/forums/threads/pet-mod-of-the-season-voting.3731591/#post-9872133, which this PR adds in preparation for the December shifts in a week.

The removal of the folder of the previous PMOTS Hoenn Gaiden and Leader's Choice JolteMons Randbats weren't included as I inexplicably got a build error when trying to remove the former, but those both can be removed. 

This should have no build errors, but if anything comes up or if you have any questions or concerns then just let me know.

File: config/formats.ts [hunk 1]
```diff
@@ -525,27 +525,28 @@ export const Formats: FormatList = [
...
-			`&bullet; <a href="https://www.smogon.com/forums/threads/hoenn-gaiden-pet-mod-of-the-season.3714737/">Hoenn Gaiden on Smogon Forums</a>`,
+			`&bullet; <a href="https://www.smogon.com/forums/threads/vaporemons-slate-1-discussion-phase.3722917/">VaporeMons</a>`,
+			`&bullet; <a href="https://docs.google.com/spreadsheets/d/1_5AwZ24dPu3-5m5yOyIO4OTPmW9OwIWXXzZ5IJZkj4c/edit?usp=sharing">Spreadsheet</a>`,
 		],
-
-		mod: 'gen3hoenngaiden',
-		ruleset: ['HG Standard', 'Freeze Clause Mod'],
-		banlist: ['Uber'],
-		unbanlist: ['Sand Veil'],
+		mod: 'vaporemons',
```
Comment (KrisXV) [inline config/formats.ts:534]:
```suggestion

		mod: 'vaporemons',
```

File: config/formats.ts [hunk 2]
```diff
@@ -525,27 +525,28 @@ export const Formats: FormatList = [
...
 	{
-		name: "[Gen 8] JolteMons Random Battle",
-		desc: `Pok&eacute;mon, items, abilities, and moves are redesigned for OU, and new items, abilities, and moves are added, all without changing base stats.`,
+		name: "[Gen 1] Modern Gen 1",
+		desc: `A Gen 1 solomod where all Pok&eacute;mon and moves from future generations are legal.`,
 		threads: [
-			`&bullet; <a href="https://www.smogon.com/forums/threads/3694234/">JolteMons</a>`,
-			`&bullet; <a href="https://docs.google.com/spreadsheets/d/149ZlQY0bJIAqfWB_233Dvbpqs3pVSHYpIoAQQkwquls/edit?usp=sharing">Spreadsheet</a>`,
+			`&bullet; <a href="https://www.smogon.com/forums/threads/gen-9-modern-gen-1.3711533/">Modern Gen 1</a>`,
 		],
```
Comment (KrisXV) [inline config/formats.ts:546]:
```suggestion
		],

```

File: data/mods/moderngen1/scripts.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	gen: 1,
+	init() {
+		for (const i in this.data.Pokedex) {
+			(this.data.Pokedex[i] as any).gender = 'N';
+			(this.data.Pokedex[i] as any).eggGroups = null;
+		}
+		const specialTypes = ['Fire', 'Water', 'Grass', 'Ice', 'Electric', 'Dark', 'Psychic', 'Dragon'];
+		let newCategory = '';
+		for (const i in this.data.Moves) {
+			if (!this.data.Moves[i]) console.log(i);
```
Comment (KrisXV) [inline data/mods/moderngen1/scripts.ts:12]:
```suggestion
```

**Review by KrisXV — CHANGES_REQUESTED** [review]
can you delete the old mod files too please?

**Comment by Yoshiblaze** [comment]
Seems that for whatever reason, when running the check online, it tries to specifically check for all the files I deleted
image

Is there any way I can add the files back to this PR and then make a separate PR where I delete the old mods?

**Comment by Yoshiblaze** [comment]
Have now made a separate pull request to remove the mods here https://github.com/smogon/pokemon-showdown/pull/9930, so this one won't have the build errors it had previously anymore.

**Pull request merged**